velox/expression/CastExpr.cpp

```diff
@@ -554,6 +554,14 @@ void CastExpr::applyPeeled(
     result = castFromDate(rows, input, context, toType);
   } else if (toType->isDate()) {
     result = castToDate(rows, input, context, fromType);
+  } else if (fromType->isUnKnown()) {
+    // Presto fail() function return type is UNKNOWN.
+    // We simply return an error as we don't know how to handle UNKNOWN values.
```

Comment:
Unknown is just null. Why do we throw casting NULL?

Reply:
I will rework the semantics. Thanks for the pointer.